tensorflow/go/attrs.go

```diff
@@ -53,92 +54,95 @@ func (op *Operation) Attr(name string) (interface{}, error) {
 func listAttribute(op *Operation, cname *C.char, meta C.TF_AttrMetadata) (interface{}, error) {
 	status := newStatus()
 
+	// NOTE: this function allocates 1 extra element in each array passed to C to
+	// allow for passing 0 length arrays.
+
 	switch meta._type {
 	case C.TF_ATTR_STRING:
-		values := make([]unsafe.Pointer, meta.list_size)
-		lengths := make([]C.size_t, meta.list_size)
-		storage := make([]C.char, meta.total_size)
+		values := make([]unsafe.Pointer, meta.list_size+1)
```

Comment:
Would it work just as well to specially handle the case for `meta.list_size == 0` instead of creating these larger lists and going through the motions? Something like:

```go
if meta.list_size == 0 {
  return make([]string, 0), nil
}
```
?

Reply:
I was previously thinking that it would force it to return an error if the attribute doesn't exist, but I realize the metadata call already does that for us. But yeah, that was my original fix. The only consideration with that is then you have to make the types match.

Reply:
Done.